core/primitives/src/types.rs

```diff
@@ -990,3 +991,23 @@ pub enum TrieCacheMode {
     /// and thus users (potentially different) are not required to pay twice for retrieval of the same node.
     CachingChunk,
 }
+
+/// Counts accessed trie nodes during tx/receipt execution for proper storage costs charging.
+#[derive(Debug)]
+pub struct TrieNodesCount {
+    /// Number of nodes read from storage or shard cache.
+    pub touches: u64,
+    /// Number of nodes read from the chunk cache.
+    pub chunk_cache_reads: u64,
+}
+
+impl Sub for TrieNodesCount {
```

Comment:
Micro nit, but, as a rule of thumb, I've found that `impl ops::Sub` usually works better (needs only one import if you impl, eg, both Sub and SubAssign)